source/ncengine/graphics2/diligent/resource/ResourceTypes.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include "Graphics\GraphicsEngine\interface\GraphicsTypes.h"
+
+#include <string>
+
+namespace nc::graphics
+{
+struct StructuredBufferResourceDesc
```

Comment:
Desc members should probably have defaults, even if they're 'invalid' states like SHADER_TYPE_UNKNOWN.